packages/angular_devkit/build_angular/src/webpack/configs/common.ts

```diff
@@ -316,78 +317,43 @@ export function getCommonConfig(wco: WebpackConfigOptions): Configuration {
   const extraMinimizers = [];
 
   if (scriptsOptimization) {
-    const TerserPlugin = require('terser-webpack-plugin');
-    const angularGlobalDefinitions = buildOptions.aot
-      ? GLOBAL_DEFS_FOR_TERSER_WITH_AOT
-      : GLOBAL_DEFS_FOR_TERSER;
-
-    // TODO: Investigate why this fails for some packages: wco.supportES2015 ? 6 : 5;
-    const terserEcma = 5;
-
-    const terserOptions = {
-      warnings: !!buildOptions.verbose,
-      safari10: true,
-      output: {
-        ecma: terserEcma,
-        // For differential loading, this is handled in the bundle processing.
-        ascii_only: !differentialLoadingMode,
-        // Default behavior (undefined value) is to keep only important comments (licenses, etc.)
-        comments: !buildOptions.extractLicenses && undefined,
-        webkit: true,
-        beautify: shouldBeautify,
-        wrap_func_args: false,
-      },
-      // On server, we don't want to compress anything. We still set the ngDevMode = false for it
-      // to remove dev code, and ngI18nClosureMode to remove Closure compiler i18n code
-      compress:
-        allowMinify &&
-        (platform === 'server'
-          ? {
-              ecma: terserEcma,
-              global_defs: angularGlobalDefinitions,
-              keep_fnames: true,
-            }
-          : {
-              ecma: terserEcma,
-              pure_getters: buildOptions.buildOptimizer,
-              // PURE comments work best with 3 passes.
-              // See https://github.com/webpack/webpack/issues/2899#issuecomment-317425926.
-              passes: buildOptions.buildOptimizer ? 3 : 1,
-              global_defs: angularGlobalDefinitions,
-              pure_funcs: ['forwardRef'],
-            }),
-      // We also want to avoid mangling on server.
-      // Name mangling is handled within the browser builder
-      mangle: allowMangle && platform !== 'server' && !differentialLoadingMode,
-    };
-
     const globalScriptsNames = globalScriptsByBundleName.map((s) => s.bundleName);
 
-    extraMinimizers.push(
-      new TerserPlugin({
-        parallel: maxWorkers,
-        extractComments: false,
-        exclude: globalScriptsNames,
-        terserOptions,
-      }),
+    if (
+      globalScriptsNames.length > 0 &&
+      (differentialLoadingMode || wco.scriptTarget < ScriptTarget.ES2015)
```

Comment:
I am not following why `differentialLoadingMode || wco.scriptTarget < ScriptTarget.ES2015` is needed here. 

- Scripts bundles are not down-levelled and they can contain a mixture of syntaxes
- We are filtering what to minimize with TerserPlugin using the `includes` option.

Reply:
This was an attempt at an optimization for the scripts.  The global scripts should only need to be ES5 when using differential loading (the scripts are shared between es5/es2015 bundles) and when targeting ES5 directly.  Otherwise they should be able to be processed by the new plugin at the specified target.  `esbuild` will error if it detects ES2015+ content when targeting ES5 so the above conditionals bypass the problem for the ES5 cases.  I think long-term the error could be beneficial since it would notify users that the scripts are not compatible with the specified output target but that would currently be a breaking change.

Alternatively, the above conditionals can be dropped and scripts can always be treated as ES5 code.  This is actually the current behavior so it may be better to retain that for now and revisit in an upcoming major release.

Reply:
I think also treating it as ES5 make sense.

Reply:
Updated it here.

Reply:
Eventually it would be nice to get rid of `terser-webpack-plugin` completely.